### PR TITLE
✨ Add Ubuntu 24 Noble to `docker-in-docker`

### PR DESCRIPTION
**Pull request by jacobwoffenden**
This pull request resolves https://github.com/devcontainers/features/issues/967 by adding `noble` to the allowed codenames

Also includes a request to include `ubuntu:noble` in the test workflows (https://github-partners.slack.com/archives/C0431R35H37/p1715616382180099?thread_ts=1715589521.243939&cid=C0431R35H37)

Signed-off-by: Jacob Woffenden <jacob@woffenden.io>

### CODE REVIEW

File: .github/workflows/test-all.yaml [hunk 1]
```diff
@@ -44,6 +44,7 @@ jobs:
           [
             "ubuntu:focal",
             "ubuntu:jammy",
+            "ubuntu:noble",
```
Comment (samruddhikhandale) [inline .github/workflows/test-all.yaml:47]: As mentioned in https://github.com/devcontainers/features/pull/969/files#r1598718691, can we use `mcr.microsoft.com/devcontainers/base:noble` instead ?

File: .github/workflows/test-pr.yaml [hunk 2]
```diff
@@ -51,6 +51,7 @@ jobs:
           [
             "ubuntu:focal",
             "ubuntu:jammy",
+            "ubuntu:noble",
```
Comment (samruddhikhandale) [inline .github/workflows/test-pr.yaml:54]: Same comment here!

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/docker-in-docker/install.sh [hunk 3]
```diff
@@ -18,8 +18,8 @@ USERNAME="${USERNAME:-"${_REMOTE_USER:-"automatic"}"}"
 INSTALL_DOCKER_BUILDX="${INSTALLDOCKERBUILDX:-"true"}"
 INSTALL_DOCKER_COMPOSE_SWITCH="${INSTALLDOCKERCOMPOSESWITCH:-"true"}"
 MICROSOFT_GPG_KEYS_URI="https://packages.microsoft.com/keys/microsoft.asc"
```
Comment (samruddhikhandale) [inline src/docker-in-docker/install.sh:20]: Can we bump minor version for the Feature? Thanks!

File: .github/workflows/test-all.yaml (hunk 1 above)
Comment (jacobwoffenden) [reply .github/workflows/test-all.yaml]: https://github.com/devcontainers/features/pull/971/commits/6b6ad16487edbe685c598ed57c0ada36995648df

File: .github/workflows/test-pr.yaml (hunk 2 above)
Comment (jacobwoffenden) [reply .github/workflows/test-pr.yaml]: https://github.com/devcontainers/features/pull/971/commits/6b6ad16487edbe685c598ed57c0ada36995648df

File: src/docker-in-docker/install.sh (hunk 3 above)
Comment (jacobwoffenden) [reply src/docker-in-docker/install.sh:20]: https://github.com/devcontainers/features/pull/971/commits/6b6ad16487edbe685c598ed57c0ada36995648df

**Review by samruddhikhandale — APPROVED** [review]
Looks wonderful, thank you!

**Pull request merged**